contracts/Remittance.sol

```diff
@@ -1,16 +1,134 @@
 pragma solidity >= 0.6.0 < 0.7.0;
 
-import "../node_modules/@openzeppelin/contracts/access/Ownable.sol";
-import "../node_modules/@openzeppelin/contracts/utils/Pausable.sol";
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
 
-import {SafeMath} from "../node_modules/@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeMath} from "./math/SafeMath.sol";
 
 contract Remittance is Ownable, Pausable {
 
-    using SafeMath for uint;
+    using SafeMath for uint256;
 
+    mapping(bytes32 => RemittanceInstance) public remittances;
 
+    struct RemittanceInstance {
+        address funder;
+        uint256 fundsOwed;
+        uint256 expiryDate;
+    }
+
+    event RemittanceFundsCreated(
+        bytes32 indexed hash,
+        address indexed funder,
+        address indexed broker,
+        uint256 amount,
+        uint256 expiryDate
+    );
+
+    event RemittanceFundsReleased(
+        bytes32 indexed hash,
+        bytes32 indexed password,
+        address indexed broker,
+        uint256 amount
+    );
+
+    event RemittanceFundsReclaimed(
+        bytes32 indexed hash,
+        address indexed funder,
+        uint256 amount
+    );
+
+    // The funder can create a remittance
+    // Hash is a sha256 hash of the broker address + recipient password
+    function create(bytes32 passwordBrokerHash, address broker, uint256 expiryDate)
+    external
+    payable
+    whenNotPaused
+    returns(bool success)
+    {
+        require(broker != msg.sender, "Caller cannot be the broker");
+        require(msg.value > 0, "Amount must be greater than 0");
+        require(broker != address(0), "Address cannot be zero");
+
+        // A 15 min margin has been added to allow for any lag
+        require(expiryDate >= block.timestamp + (24 * 60 * 60) - 1500, "Expiry less than 24h ahead");
+        checkEmptyHash(passwordBrokerHash);
+
+        RemittanceInstance storage remittanceInstance = remittances[passwordBrokerHash];
+
+        // .funder must be empty otherwise the key is already in use
+        require(remittanceInstance.funder == address(0), "Remittance already exists");
+
+        remittanceInstance.funder = msg.sender;
+        remittanceInstance.fundsOwed = msg.value;
+        remittanceInstance.expiryDate = expiryDate;
+
+        emit RemittanceFundsCreated(passwordBrokerHash, msg.sender, broker, msg.value, expiryDate);
+
+        return true;
+    }
+
+    // The broker sends the recipient's password to release the funds
+    function release(bytes32 password) external whenNotPaused returns(bool success) {
+
+        bytes32 rehash = hashPasswordBroker(password, address(msg.sender));
+
+        //Retrieve remittance
+        RemittanceInstance storage remittanceInstance = remittances[rehash];
+
+        uint256 fundsOwed = remittanceInstance.fundsOwed;
+
+        require(fundsOwed > 0, "No funds available");
+        require(block.timestamp < remittanceInstance.expiryDate, "Remittance has expired");
```

Comment:
To me this check on the expiry inside the `release` is more a spec than a requirement. The specs could well say that after the expiry, it is possible to both release and reclaim.

Reply:
Sorry. I think you had made this observation previously. I have now removed.